validation/validation_test.go

```diff
@@ -11,17 +11,21 @@ import (
 	"github.com/mrunalp/fileutils"
 	"github.com/opencontainers/runtime-tools/generate"
 	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
 )
 
 var (
 	runtime = "runc"
 )
 
 func init() {
-	runtime = os.Getenv("RUNTIME")
+	runtimeInEnv := os.Getenv("RUNTIME")
+	if runtimeInEnv != "" {
+		runtime = runtimeInEnv
+	}
 }
 
-func runtimeValidate(runtime string, g *generate.Generator) error {
+func runtimeValidate(runtime string, g *generate.Generator, args []string) error {
```

Comment:
This is not going to work, because when you test `start`, `state`, `delete` etc. you'll need to call `create` first (in most cases).  `runtimeValidate` contains all the bundle setup and teardown code, so you need a way to call multiple runtime commands inside the test, not just a way to change which single runtime command you call.  Pulling the bundle creation and teardown functionality out into separate functions seems like a more flexible approach, allowing callers to easily do whatever they want in between the two.

Reply:
Your are right, move 'create/start/state/delete' to 'args' is not enough. 
https://github.com/opencontainers/runtime-tools/pull/391/  @Mashimiao  is the way to prepare the bundle.